### PR TITLE
feat: add ability to use s3 storage

### PR DESCRIPTION
**Pull request by uchajk**
Openspout can not open xlsx writer directly on s3, so we have to write the file locally first. If s3_disk is used (see config file change), then at the end of writing into xlsx file, we upload the generated file to s3 bucket. When downloading, we again check if the s3_disk is used, and download the file accordingly.

Sometimes queue process and web process do not share the storage, and have no access to each others local storage. Therefore, I believe this feature can come in handy to others as well.

### CODE REVIEW

File: src/Jobs/DataTableExportJob.php [hunk 1]
```diff
@@ -179,6 +180,10 @@ public function handle()
         }
 
         $writer->close();
+
+        if($this->getS3Disk()) {
+            Storage::disk($this->getS3Disk())->put($filename, (new File($path))->getContent());
```
Comment (challgren) [inline src/Jobs/DataTableExportJob.php:185]: Wouldn't this exceed memory since you are reading the whole contents of the file?

**Comment by challgren** [comment]
Also related #16

**Comment by yajra** [comment]
Unfortunately, I don't have an s3 resource to check this. But per code review on a previous project where I integrated the export to s3, the code looks correct.

I also used a similar approach, didn't stream the file, and had to zip the export as it included images.

```php
Storage::disk('s3')->put('exports/'.$zipFileName, Storage::disk('tmp')->get($zipFileName));
```

@challgren do you approve of this PR?

**Review by yajra — APPROVED** [review]

**Comment by challgren** [comment]
We tested it locally and it works, we can improve it by adding tests but that's minor.

**Comment by yajra** [comment]
Released on v1.1.0 🚀 Thanks!

**Comment by challgren** [comment]
@yajra and @uchajk one thing I forgot to mention is we should fix the cleanup command too since it references the old method https://github.com/yajra/laravel-datatables-export/blob/master/src/Commands/DataTablesPurgeExportCommand.php#L34

**Comment by challgren** [comment]
Well since rackspace and digital ocean have s3 like file storage we may want to stick to the command. The user could always configure retention/purging on the bucket if they desire. But I would say the purge command should purge s3. But one thing I did notice was the temp local file is not deleted so we'll need to add that in

**Pull request merged**